src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,30 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents a Person's Contact in the address book.
+ * Parent class of Phone, Email and Address classes which extract commonalities from child classes.
+ */
+public class Contact {
+
+    public String value;
+    private boolean isPrivate;
+
+    public Contact (boolean isPrivate) {
+        this.isPrivate = isPrivate;
+    }
+
+    public boolean isPrivate() {
+        return isPrivate;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
In general, `equals()` and `hashCode()` methods should be overridden together. Find out why!